charts/datarepo-api/templates/api-deployment.yaml

```diff
@@ -18,7 +18,12 @@ spec:
         component: {{ include "datarepo-api.fullname" . }}
 ##  dynamically populates environment annotations
       annotations:
+        {{ if .Values.secretsgeneric.datarepoPassword -}}
+        checksum/secrets: {{ include (print $.Template.BasePath "/secret.yaml") . | sha256sum }}
+        {{- if .Values.podAnnotations }}
 {{ toYaml .Values.podAnnotations | indent 8 }}
+        {{- end }}
+      {{- end }}
```

Comment:
I think the {{end}} is in the wrong place.  Unless I am missing something if there is no .values.secretgeneric.datarepPassword - then the .values.podAnnotations will also not be included regardless if they exist.